chain/params.go

```diff
@@ -9,6 +9,7 @@ type Params struct {
 	Forks          *Forks                 `json:"forks"`
 	ChainID        int                    `json:"chainID"`
 	Engine         map[string]interface{} `json:"engine"`
+	Whitelists     map[string]interface{} `json:"whitelists,omitempty"`
```

Comment:
Why don't you define a struct for this `Whitelists` field?

Reply:
Maybe in the future, we will have a whitelist of bytes for something or some other type

Reply:
I'm afraid I don't understand it. What's the difference between map[string]interface{} and struct here?

Reply:
my mistake, great advice a472f298d16d2588a8e624e90f5f9e1133ba7f0c

Reply:
Fixed some more stuff 049fcf70e9089c292aba89efc9c1fb5f878e8283